R/asis-nodes.R

```diff
@@ -287,36 +287,104 @@ protect_tickbox <- function(body, ns) {
 }
 
 # FOOTNOTES --------------------------------------------------------------------
-#nocov start
-footnote_check <- function(body, ns = md_ns()) {
-  predicate <- "contains(text(), '[^') and contains(text(), ']')"
-  cascade <- glue::glue(".//md:paragraph[*[{predicate}]]")
-  xml2::xml_find_all(body, cascade, ns = ns)
-}
 
-fix_footnotes <- function(feet) {
-  char <- as.character(feet)
-  char <- gsub(
-    pattern = "([\\[][\\^]|\\])",
-    replacement = "</text><text asis='true'>\\1</text><text>",
-    x = char,
-    perl = TRUE
-  )
-  make_text_nodes(char)
+#' Protect unescaped square brackets from being escaped
+#'
+#' Commonmark allows both `[unescaped]` and `\[escaped\]` square brackets, but
+#' in the XML representation, it makes no note of which square brackets were 
+#' originally escaped and thus will escape both in the output.
+#'
+#' To fix this, we dive back into the source document and find the brackets that
+#' were originally escaped and wrap those characters in protective tags so that
+#' they do not become escaped when the XML is written. This has the benefit of 
+#' protecting footnotes and reference style links for child documents.
+#'
+#' @inheritParams resolve_anchor_links
+#' @param offset the offset in lines of the yaml header defaults to 0,
+#' indicating that the text has the yaml header stripped off. 
+#' @keywords internal
+#' @note This requires documents to be read in with `sourcepos = TRUE`
+#'
+#' @examples
+#' f <- system.file("extdata", "link-test.md", package = "tinkr")
+#' md <- yarn$new(f, sourcepos = TRUE, unescaped = FALSE)
+#' md$show()
+#' if (requireNamespace("withr")) {
+#' lnks <- withr::with_namespace("tinkr", 
```

Comment:
why the withr use here?

Reply:
Because this is an internal function and CRAN did not like my habit of documenting internal functions, so this allows me to access the internal function without using the triple colons, though, I guess I could use `asNamespace("tinkr")$protect_unescaped`